Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -552,14 +578,22 @@ public Declaration FindSelectedDeclaration(CodePane activeCodePane)
                                    item.DeclarationType != DeclarationType.ModuleOption &&
                                    item.DeclarationType != DeclarationType.Class &&
                                    item.DeclarationType != DeclarationType.Module &&
-                                   (IsSelectedDeclaration(selection, item) ||
-                                    item.References.Any(reference => IsSelectedReference(selection, reference))));
+                                   (IsSelectedDeclaration(selection, item) || item.References.Any(reference => reference.Declaration.Equals(item) && IsSelectedReference(selection, reference))))
+                    .ToList();
                 try
                 {
-                    var match = matches.SingleOrDefault() ?? AllUserDeclarations
```

Comment:
there was a bug here, `matches` *may* contain more than one matching reference/declaration, because of a resolver bug assigning a reference to the *parent* declaration in a member call. e.g. `foo.bar` gets a reference to both `foo` and `bar`, but an `IdentifierReference` is created for both against the `bar` declaration. This is really just a work-around, the real fix needs to be in the resolver. I think @autoboosh already addresses this in his PR, right @autoboosh?